jekyll/_cci2/caching.md

```diff
@@ -1,74 +1,172 @@
+
+If you're looking for a quick reference, the [cache configuration options](https://circleci.com/docs/2.0/configuration-reference/#save_cache) are documented in the main Configuration article.
+
+## Minimal `save_cache` example
+
+To save a cache of a file or directory you add the following as a step to a job in you `.circleci/config.yml`:
 
-{% raw %}
-```YAML
-- restore_cache:
-    key: projectname-{{ .Branch }}
 ```
-{% endraw %}
+- save_cache:
+    key: my-cache
+    paths:
+      - my-file.txt
+      - my-project/my-dependencies-directory
+```
+
+The path for directories is relative to the `working_directory` of your job. You can specify an absolute path if you choose.
+
+## What dependencies should I cache?
+
+The dependencies that are good to cache during a job are things like the libraries that your project depends on. For example - the libraries the are installed with `pip` in Python or `npm` for Node.js.
+
+The Docker image(s) you use for your jobs will usually have tools preinstalled that are generic for building projects using the language the image is focused on. For example the `circleci/ruby:2.4.1` image has useful tools like git, openssh-client and gzip preinstalled. So these types of tools that are not explicitly required for your project are good candidates for being stored on the Docker image.
+
+## How do I know what the path is for my dependencies?
+
+The various language dependency managers (e.g. `npm` or `pip`) each have their own paths where dependencies are installed. See our Language guides and demo projects for the specifics for your stack: <https://circleci.com/docs/2.0/demo-apps/>
+
+## Using cache keys and template values to control what is cached and restored
+
+This section is where you will learn about the art of caching. We say this because caching is a balance between getting maximum performance (using a full cache for every build) and reliability (not using an out-of-date or inappropriate cache).
+
+The way you decide on when a cache will be save or restored is to use a key which is tied an aspect of your project.
+
+When storing a new cache, the `key` value may contain special template values:
+
+Template | Description
+----|----------
+{% raw %}`{{ .Branch }}`{% endraw %} | the VCS branch currently being built
+{% raw %}`{{ .BuildNum }}`{% endraw %} | the CircleCI build number for this build
+{% raw %}`{{ .Revision }}`{% endraw %} | the VCS revision currently being built
+{% raw %}`{{ .Environment.variableName }}`{% endraw %} | the environment variable `variableName`
+{% raw %}`{{ checksum "filename" }}`{% endraw %} | a base64 encoded SHA256 hash of the given filename's contents. This should be a file committed in your repo. Good candidates are dependency manifests, such as `package.json`, `pom.xml` or `project.clj`. It's important that this file does not change between `restore_cache` and `save_cache`, otherwise the cache will be saved under a cache key different than the one used at `restore_cache` time.
```

Comment:
we don't need to specify the full details of the hash - it should be sufficient to say "a checksum of the file - so you get a new cache key if file change"